src/components/SubNav/SubNav.tsx

```diff
@@ -0,0 +1,312 @@
+            scrollableRef.current;
+
+          // Determine if we are scrolling to the left or right
+          const isScrollingRight = scrollLeft > lastScrollLeft;
+          setLastScrollLeft(scrollLeft); // Update last scroll position
+
+          // Show right fade only if scrolling to the left
+          if (!isScrollingRight) {
+            // Show right fade only if not at the end
+            const atRightEdge = scrollLeft + clientWidth >= scrollWidth;
+            setShowRightFade(!atRightEdge);
+          } else {
+            setShowRightFade(false); // Hide when scrolling right
+          }
+        }
+      }, [lastScrollLeft, setLastScrollLeft, setShowRightFade]);
+
+      useEffect(() => {
+        const refCurrent = scrollableRef.current;
+        if (refCurrent) {
+          refCurrent.addEventListener("scroll", handleScroll);
+
+          // Initial check to set the right fade effect based on the initial scroll state
+          const { scrollWidth, clientWidth, scrollLeft } = refCurrent;
+          const atRightEdge = scrollLeft + clientWidth >= scrollWidth;
+
+          // Set the right fade effect if content is scrollable and not at the right edge
+          setShowRightFade(!atRightEdge);
+
+          // If the scrollable area is already at the end, hide the right fade initially
+          if (scrollWidth <= clientWidth) {
+            setShowRightFade(false); // If the content fits in the viewport, no need for a fade
+          }
+        }
+
+        // Cleanup event listener on component unmount
+        return () => {
+          if (refCurrent) {
+            refCurrent.removeEventListener("scroll", handleScroll);
+          }
+        };
+      }, [handleScroll]);
+
+      return (
+        <Box as="nav" aria-label="Sub-navigation menu" __css={styles.base}>
+          <Box __css={styles.container}>
+            <Flex
+              alignItems="center"
+              className={className}
+              gap="1rem"
+              justify="space-between"
+            >
+              <List
+                type="ul"
+                m="0"
+                sx={{
+                  ...styles.listCleanup,
+                  ...styles.scrollableList,
+                  ...styles.primaryActions,
+                }}
+                ref={scrollableRef}
+                inline
+                noStyling
+              >
+                {primaryActions}
+                {showRightFade && (
+                  // Explicitly cast styles.fadeEffect to CSSProperties
+                  <div style={styles.fadeEffect as CSSProperties} />
+                )}
+              </List>
+              <List
+                noStyling
+                inline
+                type="ul"
+                sx={{ ...styles.secondaryActions, ...styles.listCleanup }}
+                m="0"
+                width="fit-content"
+              >
+                {secondaryActions ? secondaryActions : null}
```

Comment:
I don't think empty `ul`s should render. So you might need to wrap the entire `<List ...` with the logic check for `secondaryActions ? ....`.

What do you think @bigfishdesign13 @oliviawongnyc ?

Reply:
Yes, this condition should be moved up one level to wrap the `List` element and the `List` component should be rendered only when `secondaryActions` is `true`.